crates/node/src/networking/p2p/protocol.rs

```diff
@@ -0,0 +1,43 @@
+use std::{collections::BTreeSet, net::SocketAddr};
+
+use gevulot_node::types;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum Handshake {
+    V1(HandshakeV1),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct HandshakeV1 {
+    pub my_p2p_listen_addr: SocketAddr,
+    pub peers: BTreeSet<SocketAddr>,
+    pub http_port: Option<u16>,
+}
```

Comment:
Having ability to describe the protocol message improves quality of life several orders of magnitude, compared to working with individual values directly written into pipe. :relieved:

Reply:
Yes. For the first implementation, I didn't know how far we can go in the conception and development.